### PR TITLE
Tests for userprofile, categories, items updated

### PR DESCRIPTION
**Pull request by yiyZh**

### CODE REVIEW

**Comment by seunghk1206** [comment]
Your test cases for items and user roles are also test cases for categories... Seems like you may have to reconstruct them...

File: src/backend/categories/tests.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+    def test_default_ordering(self):
+        Category.objects.create(name="Books")
+        Category.objects.create(name="Electronics")
+        Category.objects.create(name="Appliances")
+        names = list(Category.objects.values_list("name", flat=True))
+        self.assertEqual(names, sorted(names))
+
+
+class CategorySerializerTest(TestCase):
```
Comment (seunghk1206) [inline src/backend/categories/tests.py:20]: Great edge cases!

File: src/backend/items/tests.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        listing = Listing.objects.create(
+            title="Test Book",
+            category=self.category,
+            price=10.99,
+            seller=self.user,
+        )
+        count = listing.get_purchase_request_count()
+        self.assertEqual(count, 0)  # Should return 0 initially
+
+    def test_str_method(self):
```
Comment (seunghk1206) [inline src/backend/items/tests.py:36]: The string methods are sometimes very important to monitor

File: src/backend/userprofile/tests.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        user = User.objects.create_user(username="tester", email="tester@test.com", password="pass")
+        self.assertTrue(UserProfile.objects.filter(user=user).exists())
+
+    def test_profile_str_method(self):
+        user = User.objects.create_user(username="john", email="john@test.com")
+        profile = user.profile
+        self.assertEqual(str(profile), "john@test.com's profile")
+
+
+class UserProfileAPITest(TestCase):
```
Comment (seunghk1206) [inline src/backend/userprofile/tests.py:21]: Very exquisite cases for API testing. Seems a lot of the edge cases are covered.

**Review by seunghk1206 — APPROVED** [review]
